internal/handlers/common/filter.go

```diff
@@ -53,26 +53,39 @@ func filterDocumentPair(doc *types.Document, filterKey string, filterValue any)
 		return filterOperator(doc, filterKey, filterValue)
 	}
 
+	fieldExist := true
 	docValue, err := doc.Get(filterKey)
 	if err != nil {
-		return false, nil // no error - the field is just not present
+		fieldExist = false
 	}
 
 	switch filterValue := filterValue.(type) {
 	case *types.Document:
 		// {field: {expr}}
-		return filterFieldExpr(docValue, filterValue)
+		return filterFieldExpr(docValue, filterValue, fieldExist)
```

Comment:
So pass `filterKey` instead of `docValue` as the first argument there and in all other invocations